datajoint/table.py

```diff
@@ -93,6 +97,14 @@ def declare(self, context=None):
                 "Cannot declare new tables inside a transaction, "
                 "e.g. from inside a populate/make call"
             )
+        # Enforce strict CamelCase #1150
+        if "_" in self.class_name:
```

Comment:
Please make this check stricter to test for all invalid names, e.g. a regular expression. Other valid Python names are not valid DataJoint tables names. Python is very permissive. For example, the following are examples of valid Python identifiers that should not be allowed as DataJoint table classes:

* Test2beta
* alphaBeta
* TestΣ

Reply:
@dimitri-yatsenko  How strict do we want to be about enforcing CamelCase? I wrote a function for strict enforcement on [858265c](https://github.com/datajoint/datajoint-python/pull/1152/commits/858265cad6721393b2f56dc9812b86d891c164eb), but [in CI](https://github.com/datajoint/datajoint-python/actions/runs/8381923362/job/22954561878?pr=1152), some of the class names from existing tests are no longer valid, e.g.

https://github.com/datajoint/datajoint-python/blob/f357955ab1208e7cd84afabcdead4e995aa50383/tests/schema_simple.py#L16-L25

Reply:
The regular expression was too strict. Please try this one `^[A-Z][A-Za-z0-9]*$` instead.